devlib/utils/misc.py

```diff
@@ -37,6 +37,7 @@
 import types
 import warnings
 import wrapt
+import yaml
```

Comment:
I can't remember the specifics but there is a fundamental design issue with that library. I used ruamel.yaml in Lisa instead. That might have to do with a mutable singleton used in the lib for something central that could conflict between different users in the same process.

Reply:
I remember now, it's this pattern that is bonkers:

```
import yaml
yaml.add_constructor()
```

So any other dependency can decide to change the behavior of pyyaml and everyone in the same process will get the change of behavior, devlib included. That may come from the underlying C library that this wraps but this is the best recipe for unsolvable issues.

Ruamel.yaml added a way to get some sort of context object and custom constructors are only registered against it. Every user has their own instance and no-one is stepping on each other's toes.

Reply:
Thanks for the hint. Move from `yaml` to `ruamel`.